cluster.go

```diff
@@ -195,19 +208,39 @@ func (c *clusterClient) _refresh() (err error) {
 
 	var removes []conn
 
-	c.mu.RLock()
-	for addr, cc := range c.conns {
-		fresh, ok := conns[addr]
-		if ok && (cc.replica == fresh.replica || c.rOpt == nil) {
-			conns[addr] = connrole{
-				conn:    cc.conn,
-				replica: fresh.replica,
+	if !forceRefresh {
```

Comment:
I think we are almost done. But is this `forceRefresh` flag necessary? What will happen if we don't use the flag?

Reply:
We need. Because if `conns` is same as `c.conns` after `init()` called, it doesn't update `pslots` and `rslots` in `_refresh`. we need forced refresh in `_refresh`

Reply:
Yes, you are right. I overlooked that. However, that shows a problem with the current checking logic: slot changes are not considered. We need to consider them as well, otherwise, we will miss them.

To make the comparison easy, I think we may need to keep track of some simple hashes from the `parseSlots` and `parseShards`.

Reply:
>  slot changes are not considered.

Yes, I missed that point. how about just refresh always? just refreshing is more simpler and at this point, i don't think we can get many advantages using conditional refresh.

Reply:
Ok, we can find a way to do that later.

Reply:
[3712efd](https://github.com/redis/rueidis/pull/638/commits/3712efd3a6d66db27019ae895c5bda0599ab5b3f)